### PR TITLE
Link changes

### PR DESCRIPTION
**Pull request by iXeriox**

### CODE REVIEW

File: src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        log.error("SearchByID: User: " + searchByUser.get().getUserName() + " with Token: " + searchByUser.get().getToken());
+        Optional<LinkCodeEntity> searchById = linkCodeRepo.findByToken(searchByUser.get().getToken());
+        log.error("SearchByToken: User: " + searchById.get().getUserName() + " with Token: " + searchById.get().getToken());
+
+        return event.reply()
+            .withEphemeral(true)
+            .withContent("Hello " + searchByUser.get().getUserName() + ", Your token is: " + searchByUser.get().getToken() + " .. -Insert nifty bow, here-");
+    }
+
+    private boolean addToDatabase(UUID newToken, String discordName) {
```
Comment (Adam1901) [inline src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java]: Why the bool?

File: src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        var newToken = UUID.randomUUID();
+        var discordName = event.getInteraction().getUser().getUsername();
+        if (linkCodeRepo.findById(discordName).isEmpty()) {
+            log.error("Couldn't find user in Database? Create..");
+            addToDatabase(newToken, discordName);
+            log.error("Created! with User: " + discordName + " and Token: " + newToken);
+            log.error("Let's now check to see if it's in the database and proceed that way.");
+        }
+        Optional<LinkCodeEntity> searchByUser = linkCodeRepo.findById(discordName);
```
Comment (Adam1901) [inline src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java:39]: How could this be optimised?

Optional has isempty and isPrewnet methods

File: src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    public LinkController(LinkCodeRepo linkCodeRepo) {
+        this.linkCodeRepo = linkCodeRepo;
+    }
+
+    @GetMapping(value = "/byUser/{discordUserName}", produces = "application/json")
+    public String getToken(@PathVariable("discordUserName") String discordUserName) {
+        Optional<LinkCodeEntity> getTokenById = linkCodeRepo.findById(discordUserName);
+        UUID gotToken = getTokenById.get().getToken();
+        String gotDiscordUser = getTokenById.get().getUserName();
+        return "{ \"userName\": \"" + gotDiscordUser + "\", \"token\":\"" + gotToken + "\" }";
```
Comment (Adam1901) [inline src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkController.java]: So as I mentioned in the past, spring does the object mapping

So try creating a class.called TokenResponse and return that with the values

File: src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @Autowired
+    public LinkController(LinkCodeRepo linkCodeRepo) {
+        this.linkCodeRepo = linkCodeRepo;
+    }
+
+    @GetMapping(value = "/byUser/{discordUserName}", produces = "application/json")
+    public String getToken(@PathVariable("discordUserName") String discordUserName) {
+        Optional<LinkCodeEntity> getTokenById = linkCodeRepo.findById(discordUserName);
+        UUID gotToken = getTokenById.get().getToken();
+        String gotDiscordUser = getTokenById.get().getUserName();
```
Comment (Adam1901) [inline src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkController.java]: And what if the token doesn't exist?

**Review by Adam1901 — CHANGES_REQUESTED** [review]

File: src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java (hunk 1 above)
Comment (iXeriox) [reply src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java]: No idea. Changed to "void"

**Pull request merged**